sales_team_invoiced_target_domain/model/crm_team.py

```diff
@@ -0,0 +1,30 @@
+import ast
+
+from odoo import fields, models
+from odoo.osv import expression
+
+
+class CrmTeam(models.Model):
+    _inherit = "crm.team"
+
+    def _compute_invoiced(self):
+        if not self:
+            return
+        sales_team_invoiced_domain = self.env.company.sales_team_invoiced_domain
+        if not sales_team_invoiced_domain:
+            return super()._compute_invoiced()
+        today = fields.Date.today()
+        invoiced_domain = [
+            ("move_type", "in", ["out_invoice", "out_refund", "out_receipt"]),
+            ("team_id", "in", self.ids),
+            ("date", ">=", fields.Date.to_string(today.replace(day=1))),
+            ("date", "<=", fields.Date.to_string(today)),
+        ]
+        domain_list = ast.literal_eval(sales_team_invoiced_domain)
+        invoiced_domain = expression.AND([invoiced_domain, domain_list])
+        for team in self:
+            team.invoiced = 0.0
+            moves = self.env["account.move"].search(invoiced_domain)
+            for move in moves:
+                if move.team_id == team:
+                    team.invoiced += move.amount_untaxed_signed
```

Comment:
I think this can be improved using `_read_group`; it's more efficient and can retrieve all data grouped by `team_id` in a single query. Could you update the code, please? Or is there any limitation preventing the use of `_read_group`?

Reply:
But then, if a team isn’t in the group because of the domain, we’ll still need to compute its invoiced value even if it’s 0 so the performance gain from using read_group is lost when we have to look up the team and assign its invoiced, right?

Reply:
Please check the other changes, this is the only thing left to resolve. Thx!

Reply:
Please try something like this:

```suggestion
        domain_list = ast.literal_eval(self.env.company.sales_team_invoiced_domain)
        if not domain_list:
            return super()._compute_invoiced()
        invoiced_domain = expression.AND([self._prepare_invoice_domain(), domain_list])
        team_data = self.env["account.move"]._read_group(
            invoiced_domain,
            groupby=["team_id"],
            aggregates=["amount_untaxed_signed:sum"],
            
        )
        team_dict = dict(team_data)
        for team in self:
            team.invoiced = team_dict.get(team) or 0.0
```

As you're iterating over `self`, if a team has no data in `team_dict`, the default value is 0.0. This covers your case.
Please let me know if this works for you, and if so, kindly update the test as well.

Reply:
Yep, works fine, thx!